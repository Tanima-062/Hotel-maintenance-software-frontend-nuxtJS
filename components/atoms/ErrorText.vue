<template>
  <transition name="fade">
    <p
      v-if="!!errText"
      :class="`text-alert ${NoWrap ? 'whitespace-no-wrap' : ''}`"
    >
      {{ errText }}
    </p>
  </transition>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  errText!: string

  @Prop({ default: false })
  NoWrap!: boolean
}
</script>
<style lang="sass" scoped>
.fade-enter-active, .fade-leave-active
  transition: opacity .2s

.fade-enter, .fade-leave-to /* .fade-leave-active below version 2.1.8 */
  opacity: 0
</style>
