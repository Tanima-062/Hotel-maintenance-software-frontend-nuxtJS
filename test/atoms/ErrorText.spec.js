import { shallowMount } from '@vue/test-utils'
import ErrorText from '@/components/atoms/ErrorText.vue'

describe('Atoms ErrorText Check', () => {
  test('ErrorText Test', () => {
    const props = {
      errText: 'it has some error',
    }
    const wrapper = shallowMount(ErrorText, {
      propsData: props,
    })

    // check props value
    expect(wrapper.props().errText).toBe(props.errText)

    // check error Text
    expect(wrapper.find('p').text()).toBe(props.errText)

    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
