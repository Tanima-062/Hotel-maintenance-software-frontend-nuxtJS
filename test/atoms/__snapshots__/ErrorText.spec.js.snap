// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms ErrorText Check ErrorText Test 1`] = `
<transition-stub
  name="fade"
>
  <p
    class="text-alert "
  >
    
    it has some error
  
  </p>
</transition-stub>
`;
